controllers/utils/etcd.go

```diff
@@ -33,6 +36,30 @@ func IsEtcdDisruptionAllowed(ctx context.Context, cl client.Client, log logr.Log
 		log.Info("Etcd disruption is allowed")
 		return true, nil
 	}
-	log.Info("Etcd disruption is not allowed")
+
+	// No disruptions allowed, so the only case we should remediate is that the node in question is already one of the disrupted ones
+	// The PDB doesn't disclose which node is disrupted
+	// So we have to check the etcd guard pods
+	podList := &corev1.PodList{}
+	if err := cl.List(ctx, podList, &client.ListOptions{
+		Namespace:     etcdNamespace,
+		LabelSelector: labels.SelectorFromSet(pdb.Labels),
+	}); err != nil {
+		return false, err
+	}
+	for _, pod := range podList.Items {
+		if pod.Spec.NodeName == node.Name {
+			for _, condition := range pod.Status.Conditions {
+				if condition.Type == corev1.PodReady && condition.Status == corev1.ConditionFalse {
```

Comment:
Would it be better to check if the condition status is different than True (or that it is False OR Unknown) instead of checking only if it is False?

Reply:
No, I want to be sure that etcd is already broken on this node, and `Unknown` is unclear.